### PR TITLE
[CALCITE-6024] ListSqlOperatorTable is inefficient

### PR DESCRIPTION
**Pull request by julianhyde**
ListSqlOperatorTable is inefficient if it contains a large number of operators. It currently examines the operators one by one. ReflectiveSqlOperatorTable (and its subclass, SqlStdOperatorTable) builds a map of operators by name (actually two maps, one case-sensitive and one case-insensitive).

ListSqlOperatorTable should do the same, at least in its immutable form.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/fun/OracleSqlOperatorTable.java [hunk 1]
```diff
@@ -65,15 +69,6 @@ public class OracleSqlOperatorTable extends ReflectiveSqlOperatorTable {
    * Returns the Oracle operator table, creating it if necessary.
    */
   public static synchronized OracleSqlOperatorTable instance() {
```
Comment (libenchao) [inline core/src/main/java/org/apache/calcite/sql/fun/OracleSqlOperatorTable.java:71]: `synchronized` is no longer needed.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java [hunk 2]
```diff
@@ -1364,6 +1368,7 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
   /**
    * The <code>UNNEST WITH ORDINALITY</code> operator.
    */
+  @LibraryOperator(libraries = {}) // do not include in index
```
Comment (libenchao) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java:1371]: Why do we need to remove it from the standard operators, is it a breaking change?

Comment (julianhyde) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java:1371]: The operators for `UNNEST` and `UNNEST WTH ORDINALITY` have the same name and syntax, and therefore clash when added to a map. The latter was never accessed by name anyway.

File: core/src/main/java/org/apache/calcite/sql/fun/OracleSqlOperatorTable.java (hunk 1 above)
Comment (julianhyde) [reply core/src/main/java/org/apache/calcite/sql/fun/OracleSqlOperatorTable.java:71]: good catch; will fix

**Pull request closed without merging**